restlicsharpclient/restliclienttest/RequestBuilderBaseTests.cs

```diff
@@ -0,0 +1,134 @@
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+using System.Collections.Generic;
+using System.IO;
+
+using com.linkedin.restli.test.api;
+using restlicsharpclient.restliclient.request.builder;
+using restlicsharpclient.restliclient.request;
+using restlicsharpclient.restliclient.response;
+
+namespace restlicsharpclient.restliclienttest
+{
+    [TestClass]
+    public class RequestBuilderBaseTests
+    {
+        public static string baseUrl = "test";
+
+        [TestMethod]
+        public void SetHeader()
+        {
+            GetRequestBuilder<int, Greeting> b = new GetRequestBuilder<int, Greeting>(baseUrl);
+            b.AddHeader("header1", "badvalue1");
+            b.SetHeader("header1", new List<string>() { "value1" });
+            b.AddHeader("header2", "value2");
+            b.AddHeader("header2", "value22");
+
+            CollectionAssert.AreEqual(new List<string>() { "value1" }, b.GetHeader("header1"));
+            CollectionAssert.AreEqual(new List<string>() { "value2", "value22" }, b.GetHeader("header2"));
+        }
+
+        [TestMethod]
+        public void AddHeader()
+        {
+            GetRequestBuilder<int, Greeting> b = new GetRequestBuilder<int, Greeting>(baseUrl);
+            b.AddHeader("header1", "value1");
+            b.AddHeader("header2", "value2");
+            b.AddHeader("header2", "value22");
+
+            CollectionAssert.AreEqual(new List<string>() { "value1" }, b.GetHeader("header1"));
+            CollectionAssert.AreEqual(new List<string>() { "value2", "value22" }, b.GetHeader("header2"));
+        }
+
+        [TestMethod]
+        public void SetParam()
+        {
+            GetRequestBuilder<int, Greeting> b = new GetRequestBuilder<int, Greeting>(baseUrl);
+            Assert.IsFalse(b.HasParam("param1"));
+
+            b.SetParam("param1", 123);
+            Assert.IsTrue(b.HasParam("param1"));
+            Assert.AreEqual(123, b.GetParam("param1"));
+
+            b.SetParam("param1", "value1");
+            b.SetParam("param2", "value2");
+            Assert.AreEqual("value1", b.GetParam("param1"));
+            Assert.IsTrue(b.HasParam("param2"));
+            Assert.AreEqual("value2", b.GetParam("param2"));
+
+            b.SetParams(new Dictionary<string, object>() { { "param1", "new value1" } });
+            Assert.IsTrue(b.HasParam("param1"));
+            Assert.AreEqual("new value1", b.GetParam("param1"));
+            Assert.IsTrue(b.HasParam("param2"));
+            Assert.AreEqual("value2", b.GetParam("param2"));
+
+            b.ClearParams();
+            Assert.IsFalse(b.HasParam("param1"));
+            Assert.IsFalse(b.HasParam("param2"));
+        }
+
+        [TestMethod]
+        public void ReqSetParam()
```

Comment:
This has no difference from SetParam test. In Java, we have a different method `setReqParam` which will throw error if the passed value is null. Since we don't have this method here, no need to have this test anymore.